tests/core/test_multi_derived_signal.py

```diff
@@ -0,0 +1,167 @@
+import asyncio
+from typing import Generic, TypeVar, cast
+from unittest.mock import ANY, call
+
+import numpy as np
+import pytest
+from bluesky.protocols import Reading
+from typing_extensions import TypedDict
+
+from ophyd_async.core import (
+    Array1D,
+    AsyncStatus,
+    DerivedSignalCreator,
+    Device,
+    DTypeScalar_co,
+    Transform,
+    soft_signal_rw,
+)
+from ophyd_async.sim import SimMotor
+from ophyd_async.testing import (
+    assert_describe_signal,
+    assert_reading,
+    assert_value,
+    get_mock,
+)
+
+F = TypeVar("F", float, Array1D[np.float64])
+
+
+def convert_to_type_of(arr: Array1D[DTypeScalar_co], *need_types: F) -> F:
+    types = {type(need_type) for need_type in need_types}
+    if len(types) != 1:
+        msg = f"All need_types must be the same type, got {types}"
+        raise ValueError(msg)
+    return cast(F, arr if types.pop() == np.ndarray else float(arr))
+
+
+class TwoJackRaw(TypedDict, Generic[F]):
+    jack1: F
+    jack2: F
+
+
+class TwoJackDerived(TypedDict, Generic[F]):
+    height: F
+    angle: F
+
+
+class TwoJackTransform(Transform):
+    distance: float
+
+    def raw_to_derived(self, *, jack1: F, jack2: F) -> TwoJackDerived[F]:
+        diff = jack2 - jack1
+        return TwoJackDerived(
+            height=jack1 + diff / 2,
+            # need the case as returns numpy float rather than float64, but this
+            # is ok at runtime
+            angle=convert_to_type_of(np.arctan(diff / self.distance), jack1),
+        )
+
+    def derived_to_raw(self, *, height: F, angle: F) -> TwoJackRaw[F]:
+        diff = convert_to_type_of(np.tan(angle) * self.distance, height)
+        return TwoJackRaw(
+            jack1=height - diff / 2,
+            jack2=height + diff / 2,
+        )
+
+
+class MirrorDerived(TypedDict):
+    x: float
+    roll: float
+
+
+class Mirror(Device):
+    def __init__(self, name=""):
+        # Raw signals
+        self.x1 = SimMotor()
+        self.x2 = SimMotor()
+        # Parameter
+        self.x1_x2_distance = soft_signal_rw(float, initial_value=1)
+        # Derived signals
+        self._creator = DerivedSignalCreator(
+            TwoJackTransform,
+            self.set,
+            jack1=self.x1,
+            jack2=self.x2,
+            distance=self.x1_x2_distance,
+        )
+        self.x = self._creator.derived_signal_rw(float, "height", "x")
+        self.roll = self._creator.derived_signal_rw(float, "angle", "roll")
+        super().__init__(name=name)
+
+    @AsyncStatus.wrap
+    async def set(self, derived: MirrorDerived) -> None:
```

Comment:
Should: Is this designed to be externally callable? In which case the device should be `Movable`? Or if it's not meant to be externally callable rename from `set` for clarity